BricksAndHearts.UnitTests/ControllerTests/Property/PropertyControllerTests.cs

```diff
@@ -272,4 +269,42 @@ public void AddNewPropertyCancelPost_WithNoAddInProgress_RedirectsToViewProperti
         A.CallTo(() => PropertyService.DeleteProperty(fakePropertyDbModel)).MustNotHaveHappened();
         result.Should().BeOfType<RedirectToActionResult>().Which.ActionName.Should().Be("ViewProperties");
     }
-}
+
+    [Fact]
+    public void ViewProperty_WithNonExistingProperty_Returns404ErrorPage()
+    {
+        // Arrange
+        A.CallTo(() => PropertyService.GetPropertyByPropertyId(1)).Returns(null);
+        
+        var landlordUser = CreateLandlordUser();
+        MakeUserPrincipalInController(landlordUser, UnderTest);
+        
+        // Act
+        var result = UnderTest.ViewProperty(1);
+        
+        // Assert
+        A.CallTo(() => PropertyService.GetPropertyByPropertyId(1)).MustHaveHappened();
+        result.Should().BeOfType<RedirectToActionResult>().Which.ActionName.Should().Be("Error");
+        result.As<RedirectToActionResult>().RouteValues.Should().Contain("status",404);
+    }
+    
+    [Fact]
+    public void ViewProperty_WithExistingProperty_ReturnViewPropertyView()
+    {
+        // Arrange
+        var model = CreateExamplePropertyDbModel();
+        model.Id = 1;

```

Comment:
The model ID is already 1, it's created by the method above :) (you can delete this line)